sdk/formrecognizer/Azure.AI.FormRecognizer/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,159 @@
+# Troubleshoot Azure Form Recognizer client library issues
+
+This troubleshooting guide contains instructions to diagnose frequently encountered issues while using the Azure Form Recognizer client library for .NET.
+
+## Table of Contents
+* [Troubleshooting errors and exceptions](#troubleshooting-errors-and-exceptions)
+    * [Handling RequestFailedException](#handling-requestfailedexception)
+    * [Build model error](#build-model-error)
+       * [Invalid training data set](#invalid-training-data-set)
+       * [Invalid SAS URL](#invalid-sas-url)
+    * [Generic error](#generic-error)
+    * [The modelId must be a valid GUID](#the-modelid-must-be-a-valid-guid)
+* [Unexpected time to build a custom model](#unexpected-time-to-build-a-custom-model)
+* [Enable HTTP request/response logging](#enable-http-requestresponse-logging)
+
+## Troubleshooting errors and exceptions
+
+### Handling RequestFailedException
+The client methods in this SDK raise a [RequestFailedException](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/RequestFailedException.cs) on request failure. The `RequestFailedException` raised by the Azure Form Recognizer client library includes detailed error response information that provides useful insights into what went wrong and includes corrective actions to fix common issues. This error information can be found inside the `Message` property of the `RequestFailedException` instance.
```

Comment:
These are the general errors each client gets. Should this be in the general troubleshooting section instead? Also, the wording is a bit odd.

Consider saying something like Azure.AI.FormRecognizer library or FormRecognizerClient raises exceptions defined in the Azure.Core library for instance RequestFailedException. Including a readme section or linking to a section that talks more about RequestFailedException will also be useful.

Reply:
> These are the general errors each client gets. Should this be in the general troubleshooting section instead?
Good point. I'll move it to the parent "Troubleshooting errors and exceptions" section. It can serve as a sort of introduction to the following subsections.

Reply:
> I think you should also talk about how to enable client logging and general authentication errors as well.

There's a section for client logging at the end of the document, also similar to what we have in our README.

Usually we don't see customer issues related to authentication in FR, so they may be out of the scope of the troubleshooting guide (focus on frequent issues). Do we usually see them in other services?

Reply:
Thank you Caio. I am not sure if other SDKs see a lot of general errors and I am not sure if we have consistency across our SDKs on TSG formats so if we do not see errors - having a section in the end should suffice.